content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1433 @@
+    appWindow.webContents.send('send-xrp-transaction-finish', result)
+  })
+})
+
+// Step 8 code additions - start
+ipcMain.on('destination-account-change', (event, destinationAccount) => {
+  verify(destinationAccount, client).then((result) => {
+    appWindow.webContents.send('update-domain-verification-data', result)
+  })
+})
+// Step 8 code additions - end
+```
+
+3. Modify `view/preload.js` and add the following two functions to `'electronAPI'`:
+
+```javascript
+onDestinationAccountChange: (callback) => {
+  ipcRenderer.send('destination-account-change', callback)
+},
+onUpdateDomainVerificationData: (callback) => {
+  ipcRenderer.on('update-domain-verification-data', callback)
+}
+```
+
+Finally, the code decodes the account's `Domain` field, if present, and performs domain verification using the method imported above.
+
+4. Update the view logic - in `view/template.html` add the following lines just before the `<input>` element with `id="input-destination-address`:
+
+```html
+<div class="input-group mb-3">
+    <div class="accountVerificationIndicator">
+        <span>Verification status:</span>
+    </div>
+    <input type="text" class="form-control" placeholder="rn95xwUymaMyzAKnZUGuynjZ6qk9RzV4Q7" id="input-destination-address">
+    <!-- Step 8 code additions - start -->
+    <span class="input-group-text">To (Address)</span>
+    <!-- Step 8 code additions - start -->
+</div>
+```
+
+5. Lastly, modify the renderer as described below:
+
+```javascript
+modalButton.addEventListener('click', () => {
+    modalDialog.show()
+})
+
+// Step 8 code additions - start
+const accountVerificationEl = document.querySelector('.accountVerificationIndicator span')
+// Step 8 code additions - end
+
+const destinationAddressEl = document.getElementById('input-destination-address')
+const destinationTagEl = document.getElementById('input-destination-tag')
+const amountEl = document.getElementById('input-xrp-amount')
+const sendXrpButtonEl = document.getElementById('send-xrp-submit-button')
+
+// Step 8 code additions - start
+destinationAddressEl.addEventListener('input', (event) => {
+    window.electronAPI.onDestinationAccountChange(destinationAddressEl.value)
+})
+
+window.electronAPI.onUpdateDomainVerificationData((_event, result) => {
+    accountVerificationEl.textContent = `Domain: ${result.domain || 'n/a'} Verified: ${result.verified}`
+})
+// Step 8 code additions - end
+
+sendXrpButtonEl.addEventListener('click', () => {
+    modalDialog.hide()
+    const destinationAddress = destinationAddressEl.value
+    const destinationTag = destinationTagEl.value
+    const amount = amountEl.value
+
+    window.electronAPI.onClickSendXrp({destinationAddress, destinationTag, amount})
+})
+```
+
+
+The updated preloader `view/8_preloader.js` is also modified the same way by adding the following two event listeners:
+
+`view/8_preload.js`
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/8_preload.js", language="js", lines="33-38") }}
```

Comment:
I'm confused what the point of these lines are, I think they can be deleted.

Reply:
Seems to be a leftover doublet after I restructured the Steps to handle files in the same order

Reply:
Done